version: "3.8"
networks:
  pulsar:
    driver: bridge
  db_network:
    driver: bridge

services:
  # Apache Pulsar en modo standalone
  pulsar:
    image: apachepulsar/pulsar:latest
    container_name: pulsar
    profiles: ["pulsar"]
    hostname: broker
    restart: on-failure
    networks:
      - pulsar
    ports:
      - "6650:6650"
      - "8080:8080"
    environment:
      - PULSAR_MEM=-Xms512m -Xmx1g
    command: bin/pulsar standalone
    deploy:
      resources:
        limits:
          cpus: "1.0"
          memory: "1.5G"

  # Servicios de Salud Tech
  salud_tech:
    build:
      context: .
      dockerfile: src/salud_tech/Dockerfile
    container_name: salud_tech
    hostname: salud_tech
    profiles: ["salud_tech"]
    networks:
      - pulsar
      - db_network
    environment:
      - FLASK_APP=src.salud_tech.api
      - FLASK_ENV=development
      - FLASK_DEBUG=1
      - HOST_ADDRESS=salud_tech
      - PULSAR_ADDRESS=broker
      - BROKER_HOST=broker
    ports:
      - "5000:5000"

  salud_tech_db:
    image: postgres:16-alpine
    profiles: ["db"]
    environment:
      POSTGRES_USER: salud_tech
      POSTGRES_PASSWORD: salud_tech_123
      POSTGRES_DB: rabbit_salud_tech
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U salud_tech"]
      interval: 2s
      timeout: 5s
      retries: 5
    networks:
      - db_network
